lib/language_pack/ruby.rb

```diff
@@ -117,6 +118,66 @@ def compile
     raise e
   end
 
+  def patch_libpq
+    # Check for existing libraries
+    out = run!("ls /usr/lib/x86_64-linux-gnu/ | grep libpq")
+    out.each_line do |line|
+      version = line.sub(/libpq\.so\./, '')
+      next unless version.match?(/\Ad/) # avoid libpq.so.libpq.a
+      return if Gem::Version.new(version) >= Gem::Version("5.12")
+    end
+
+    case ENV['STACK']
+    when 'heroku-16'
+      pkg = "libpq5_12.1-1.pgdg16.04+1_amd64.deb"
```

Comment:
Are old package point release versions ever deleted from the APT repo? Glancing at https://apt.postgresql.org/pub/repos/apt/pool/main/p/postgresql-12/ and similar from older major releases it seems the answers is "no", but it's perhaps worth checking in more detail, since this has bitten us in the past with Canonical's APT repos.

Reply:
On Ubuntu's PPAs, they are, yup, which is what's bitten us in the past.